slsSupportLib/include/sls/bit_utils.h

```diff
@@ -18,4 +19,76 @@ template <typename T> std::vector<int> getSetBits(T val) {
     }
     return set_bits;
 }
+
+class RegisterAddress {
+  private:
+    uint32_t addr_{0};
+
+  public:
+    constexpr RegisterAddress() noexcept = default;
+    explicit RegisterAddress(uint32_t address) : addr_(address) {}
+    explicit RegisterAddress(const std::string &address);
+
+    std::string str() const;
+    operator uint32_t() const noexcept { return addr_; }
+
+    constexpr bool operator==(const RegisterAddress &other) const {
+        return (addr_ == other.addr_);
+    }
+    constexpr bool operator!=(const RegisterAddress &other) const {
+        return (addr_ != other.addr_);
+    }
+} __attribute__((packed));
+
+class BitPosition {
+  private:
+    RegisterAddress addr_{0};
+    int bitPos_{0};
+
+  public:
+    constexpr BitPosition() noexcept = default;
+    BitPosition(RegisterAddress address, int bitPosition);
+    std::string str() const;
+
+    RegisterAddress address() const noexcept { return addr_; }
+    int bitPosition() const noexcept { return bitPos_; }
+    void setAddress(RegisterAddress address) noexcept { addr_ = address; }
+    void setBitPosition(int bitPosition) noexcept { bitPos_ = bitPosition; }
+
+    constexpr bool operator==(const BitPosition &other) const {
+        return (addr_ == other.addr_ && bitPos_ == other.bitPos_);
+    }
+    constexpr bool operator!=(const BitPosition &other) const {
+        return !(*this == other && bitPos_ == other.bitPos_);
+    }
+} __attribute__((packed));
+
+class RegisterValue {
+  private:
+    uint32_t value_{0};
+
+  public:
+    constexpr RegisterValue() noexcept = default;
+    explicit constexpr RegisterValue(uint32_t value) noexcept : value_(value) {}
+    explicit RegisterValue(const std::string &value);
+
+    std::string str() const;
+    operator uint32_t() const noexcept { return value_; }
+
+    RegisterValue &operator|=(uint32_t rhs) noexcept {
```

Comment:
Do we want to allow this with an uint32_t and not only with other RegisterValues?